### DISCUSSION TITLE
Consider settings startup environment variables in runtimeConfig.json

### DISCUSSION THREAD

**Discussion by msedi** [dotnet/runtime#63783] [Ideas]
Currently there exists no way to configure environment variables on the startup of a .NET desktop application. This is especially relevant when you want to configure environment variables that control the .NET runtime behavior `DOTNET_Thread_UseAllCpuGroups`.

The current solution is to create a starter batch file that sets the parameters. `launchSettings.json` already has a possibility to startup the debug session with environment variables, **so I would suggest to introduce the same concept in `runtimeConfig.json`**

```
    "environmentVariables": {
        "DOTNET_Thread_UseAllCpuGroups": "true"
      }
```

**Comment by huoyaoyuan** [comment]
> This is especially relevant when you want to configure environment variables that control the .NET runtime behavior

Doesn't it have an alternative form in runtime config?

**Reply by msedi** [reply]
I looked in up in the schema definition but couldn't find any docs about this. Moreover the problem is that I need to set in the runtimeConfig.template.json which seems to allow only settings configProperties (like this example here):

```
 "configProperties": {
      "System.GC.Concurrent": false,
      "System.Threading.ThreadPool.MinThreads": 4,
      "System.Threading.ThreadPool.MaxThreads": 25
    }
```

Maybe I'm wrong.
